src/mria_config.erl

```
@@ -259,6 +271,20 @@ consistency_check() ->
             error(unsupported_otp_version);
          _ ->
             ok
+    end,
+    ExtraMnesiaDiagnosticChecks = get_extra_mnesia_diagnostic_checks(),
+    AllValidChecks =
+        lists:all(fun({_Name, CheckFn}) when is_function(CheckFn, 0) -> true;
+                     (_) -> false
+                  end,
+                  ExtraMnesiaDiagnosticChecks),
+    case AllValidChecks of
+        true ->
+            ok;
+        false ->
+            ?LOG(critical, "Configuration error: extra mnesia diagnostic "
+                           "checks must be of type [{any(), any(), fun(() -> any())}]", []),
+            error(bad_mnesia_diagnostic_function)
```

Can this error abort the recovery? We probably don't want that, given that this code runs when mnesia is a "strange" state. Maybe it even makes sense to catch all exceptions and exits insted.

---

:heavy_check_mark: